seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/reader/AbstractReadStrategy.java

```diff
@@ -153,15 +151,10 @@ public List<String> getFileNamesByPath(HadoopConf hadoopConf, String path) throw
                 }
             }
         }
-
-        if (fileNames.isEmpty()) {
-            throw new FileConnectorException(
-                    FileConnectorErrorCode.FILE_LIST_EMPTY,
-                    "The target file list is empty,"
-                            + "SeaTunnel will not be able to sync empty table, "
-                            + "please check the configuration parameters such as: [file_filter_pattern]");
+        if (this.fileNames.isEmpty()) {
+            log.error("The current directory is empty " + path);
+            this.fileNames.add(path);
```

Comment:
为啥还把文件夹的名称加进去，fileNames里面装的是可阅读的文件的名称，只要保证这个方法返回的集合里的文件不为空，如若为空则抛出异常就可以。

Reply:
> 为啥还把文件夹的名称加进去，fileNames里面装的是可阅读的文件的名称，只要保证这个方法返回的集合里的文件不为空，如若为空则抛出异常就可以。

cc @zhilinli123

Reply:
looks the up-to-date code not add the path, just logger the error info. @ponyliuh 
And we can't throw exception when size is empty. because if the table is empty, we still want it run well, just sync nothing not get an exception.
![image](https://github.com/apache/seatunnel/assets/28497357/705b1801-5d9a-437d-8984-f2a5affdb726)